File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key = lambda x : x[1])

        arrows = 1
        current_end = points[0][1]

        for start, end in points:
            if current_end < start:
                arrows += 1
                current_end = end

        return arrows

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cs
public class Solution {
    public int FindMinArrowShots(int[][] points) 
    {
        points = points.OrderBy(x => x[1]).ToArray();

        var arrows = 1;
        var currentEnd = points[0][1];

        foreach (var point in points)
        {
            var start = point[0];
            var end = point[1];
            if (currentEnd >= start) continue;
            arrows++;
            currentEnd = end;
        }

        return arrows;
    }
}